packages/std/src/addresses.rs

```diff
@@ -42,49 +53,111 @@ impl Addr {
     /// let address = Addr::unchecked("foobar");
     /// assert_eq!(address, "foobar");
     /// ```
-    pub fn unchecked<T: Into<String>>(input: T) -> Addr {
-        Addr(input.into())
+    pub const fn unchecked(input: &str) -> Addr {
+        let input_bytes = input.as_bytes();
+        let mut data = [0u8; ADDR_MAX_LENGTH];
+        let mut i = 0;
+        while i < input_bytes.len() {
+            data[i] = input_bytes[i];
+            i += 1;
+        }
```

Comment:
```suggestion
        data[..input.len()].clone_from_slice(input.as_bytes());
```
I think we can do this here?

Reply:
Here we are very limited in what is available because it must be `const fn` compatible. We can't even rewrite the `while` loop to a `for i in 0...input_bytes.len()` loop. `clone_from_slice` is not a `const fn`, so it cannot be used.

Reply:
Thinking about optimizing it, I looked at the implementation of eg. https://doc.rust-lang.org/core/ptr/fn.copy_nonoverlapping.html but that just point to [rust intrinsics](https://doc.rust-lang.org/src/core/intrinsics.rs.html#1844-1847), so I guess this is the best we can do

Reply:
Wow – seems like we can use `copy_nonoverlapping` directly because it became const (https://github.com/rust-lang/rust/pull/79684). For some reason this is not shown in the docs yet. Will try.